ddspipe_core/test/unittest/logging/dds_consumer/DdsLogConsumerTest.cpp

```diff
@@ -153,7 +142,7 @@ TEST_F(DdsLogConsumerTest, publish_logs)
         utils::Log::Flush();
 
         // Wait for the subscriber to receive the message
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        ASSERT_TRUE(reader_->wait_for_unread_message(test::logging::MAX_WAITING_TIME));
```

Comment:
The main difference I see is that with this code the reader is waiting for only one new message to continue the tests, while with the old code, the reader could receive many messages. I assume that this is the expected behaviour but I just want you to confirm that.

Reply:
Correct. Since the test itself is initializing the `Log` with its own filters, only its own logs should be displayed. Furthermore, the test is no longer sleeping for an arbitrary amount of time and checking if it has received any logs, but waiting until it knows for sure it has.